### ISSUE TITLE
False positive for `unused_parens` in `x as (T) < y`

### ISSUE THREAD

**Issue by lukas-code** [rust-lang/rust#106413]
Given the following code: [Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=ed59e265f862c951739a630fccd675b6)

```rust
fn main() {
    let _ = 1 as (i32) < 2;
}
```

The current output is:

```
warning: unnecessary parentheses around type
 --> src/main.rs:2:18
  |
2 |     let _ = 1 as (i32) < 2;
  |                  ^   ^
  |
  = note: `#[warn(unused_parens)]` on by default
help: remove these parentheses
  |
2 -     let _ = 1 as (i32) < 2;
2 +     let _ = 1 as i32 < 2;
  |
```

But with the suggestion applied the compiler displays an error with a different suggestion:

```
error: `<` is interpreted as a start of generic arguments for `i32`, not a comparison
 --> src/main.rs:2:22
  |
2 |     let _ = 1 as i32 < 2;
  |                      ^ -- interpreted as generic arguments
  |                      |
  |                      not interpreted as comparison
  |
help: try comparing the cast value
  |
2 |     let _ = (1 as i32) < 2;
  | 
```

Ideally the compiler should not output a warning and treat `x as (T) < y` and `(x as T) < y` as identical.

@rustbot label D-incorrect A-suggestion-diagnostics

**Comment by notJoon** [comment]
I think it's relate with #22644 and #60146

**Comment by scottmcm** [comment]
It would be great if it could give
```text
  |
2 -     let _ = 1 as (i32) < 2;
2 +     let _ = (1 as i32) < 2;
  |
```
since parenthesizing the `as` expression instead avoids the potential "is that a tuple?" confusion of `(i32)`.

**Comment by kadiwa4** [comment]
Duplicate of #80636.
